cmd/clusterctl/examples/openstack/provider-component/user-data/centos/templates/master-user-data.sh

```diff
@@ -78,7 +78,7 @@ function install_configure_docker () {
 install_configure_docker
 
 # Get docker cgroup driver for kubelet configuration
-CG_DRIVER=$(docker info --format '{{.CgroupDriver}}')
+CG_DRIVER=$(docker info --format "\{\{.CgroupDriver\}\}")
```

Comment:
Do you know why CentOS and Ubuntu has different behavior? The script do works well in Ubuntu.

Reply:
the ubuntu using cgroupfs, so it did not check the docker cgroup driver configuration.

Reply:
OK, but the question is why the **old** script works for Ubuntu but not for CentOS?

```
root@gyliu-dev1:~# CG_DRIVER=$(docker info --format '{{.CgroupDriver}}')
root@gyliu-dev1:~# echo $CG_DRIVER
cgroupfs
root@gyliu-dev1:~# CG_DRIVER=$(docker info --format "\{\{.CgroupDriver\}\}")
root@gyliu-dev1:~# echo $CG_DRIVER
\{\{.CgroupDriver\}\}
```

```
root@gyliu-dev1:~# cat /etc/*release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.4 LTS"
NAME="Ubuntu"
VERSION="16.04.4 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.4 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

Reply:
the issue was caused by the text/template(https://golang.org/pkg/text/template/#Template) can not parse this and will failed at https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/pkg/cloud/openstack/machine/machineScript.go#L61.

add `\` here was just used to pass the template parse.

Reply:
/approve

/cc @jichenjc

Reply:
@hchenxa I think I tried centos before and it works well, as @gyliu513 and you discussed, this should be some change in order to trigger the failure ? the change looks good but I want to make sure we know what's the root cause... thanks

Reply:
thanks @jichenjc and @gyliu513 's comments, I mark this to WIP first as there still have some issue and will try to get them solved today.

Reply:
looks like we do not need to check the cgroup driver as kubeadm will help to do this check, refer https://github.com/kubernetes/kubernetes/pull/64347/files

I remove the logic of checking cgroup driver in user-data, cc @gyliu513 @jichenjc

Reply:
The fix lgtm, leave to @jichenjc for the lgtm label.

Reply:
thanks for the reference and update @hchenxa